plugin/kotel/carrier_test.go

```diff
@@ -0,0 +1,141 @@
+package kotel
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+func TestNewRecordCarrier(t *testing.T) {
+	type args struct {
+		record *kgo.Record
+	}
+	tests := []struct {
+		name string
+		args args
```

Comment:
Why not `record *kgo.Record`, and remove args & the args type?

Same thought below, w.r.t. the fields type as well -- although I think I know why args and fields exist, it seems a bit unnecessary here and is adding a bit of visual noise that's a bit difficult to parse